src/server/hotspot_partition_calculator.cpp

```diff
@@ -98,5 +104,60 @@ void hotspot_partition_calculator::data_analyse()
     }
 }
 
+/*static*/ void
+hotspot_partition_calculator::send_hotkey_detect_request(const std::string &app_name,
+                                                         const int partition_index,
+                                                         const hotkey_detect_type hotkey_type,
+                                                         const hotkey_detect_action action)
+{
+    dsn::apps::hotkey_detect_request request;
+    request.type = (hotkey_type == hotkey_detect_type::WRITE_HOTKEY_DETECT)
+                       ? dsn::apps::hotkey_type::WRITE
+                       : dsn::apps::hotkey_type::READ;
+    request.action = (action == hotkey_detect_action::STOP_HOTKEY_DETECT)
+                         ? dsn::apps::hotkey_detect_action::STOP
+                         : dsn::apps::hotkey_detect_action::START;
+    ddebug_f("{} {} hotkey detection in {}.{}",
+             (action == hotkey_detect_action::STOP_HOTKEY_DETECT) ? "Stop" : "Start",
+             (hotkey_type == hotkey_detect_type::WRITE_HOTKEY_DETECT) ? "write" : "read",
+             app_name,
+             partition_index);
+    dsn::rpc_address meta_server;
+    meta_server.assign_group("meta-servers");
+    std::vector<dsn::rpc_address> meta_servers;
+    replica_helper::load_meta_servers(meta_servers);
+    for (const auto &address : meta_servers) {
+        meta_server.group_address()->add(address);
+    }
+    auto cluster_name = dsn::replication::get_current_cluster_name();
+    auto resolver = partition_resolver::get_resolver(cluster_name, meta_servers, app_name.c_str());
+    dsn::task_tracker tracker;
+    resolver->call_op(
+        RPC_DETECT_HOTKEY,
+        request,
+        &tracker,
+        [app_name, partition_index](
+            dsn::error_code error, dsn::message_ex *request, dsn::message_ex *response) {
```

Comment:
Call via rpc_holder.

hotkey_detect_rpc rpc(std::move(req), RPC_DETECT_HOTKEY, 10_s, partition_index);
rpc.call(resolver, nullptr, []() {})->wait();

Reply:
`hotkey_detect_rpc` will be proposed in future PR, I can add a TODO to refactor it here

Reply:
Why leave this simple refactoring to the future? Any issue that relates to it? It will change little.